src/backend/distributed/README.md

```diff
@@ -0,0 +1,2472 @@
+    date_of_birth DATE,
+    country_code VARCHAR(3)
+);
+SELECT create_distributed_table('users_table', 'user_id');
+
+-- Distributed Table: Orders Table
+CREATE TABLE orders_table (
+    order_id bigserial,
+    user_id BIGINT REFERENCES users_table(user_id),
+    product_id BIGINT,
+    order_date TIMESTAMPTZ,
+    status VARCHAR(20)
+);
+SELECT create_distributed_table('orders_table', 'user_id');
+
+
+-- Distributed Table: Products Table
+CREATE TABLE products_table (
+    product_id bigserial PRIMARY KEY,
+    product_name VARCHAR(100),
+    category_id INT,
+    price NUMERIC(10, 2)
+);
+SELECT create_distributed_table('products_table', 'product_id');
+
+-- Reference Table: Country Codes
+CREATE TABLE country_codes (
+    country_code VARCHAR(3) PRIMARY KEY,
+    country_name VARCHAR(50)
+);
+
+-- Reference Table: Order Status
+CREATE TABLE order_status (
+    status VARCHAR(20) PRIMARY KEY,
+    description TEXT
+);
+SELECT create_reference_table('order_status');
+
+-- Reference Table: Product Categories
+CREATE TABLE product_categories (
+    category_id INT PRIMARY KEY,
+    category_name VARCHAR(50)
+);
+SELECT create_reference_table('product_categories');
+```
+
+## Fast Path Router Planner
+
+The Fast Path Router Planner is specialized in optimizing queries that are both simple in structure and certain to touch a single shard. Importantly, it targets queries on a single shard distributed, citus local or reference tables. This does not mean the planner is restricted to trivial queries; it can handle complex SQL constructs like `GROUP BY`, `HAVING`, `DISTINCT`, etc., as long as these operate on a single table and involve an equality condition on the distribution key (`distribution_key = X`). The main SQL limitation for fast path distributed query planning is the subquery/CTE support. Those are left to the next planner: Router planner.
+
+The aim of this planner is to avoid relying on PostgreSQL's standard_planner() for planning, which performs unnecessary computations like cost estimation, irrelevant for distributed planning. Skipping the standard_planner has significant performance gains for OLTP workloads. By focusing on "shard-reachable" queries, the Fast Path Router Planner is able to bypass the need for more computationally expensive planning processes, thereby accelerating query execution.
+
+### Main C Functions Involved:
+
+- `FastPathRouterPlan()`: The primary function for creating the fast-path query plan.
+- `FastPathRouterQuery()`: Validates if a query is eligible for fast-path routing by checking its structure and the WHERE clause.
+
+With  set client_min_messages to debug4; you should see the following in the DEBUG messages: "DEBUG:  Distributed planning for a fast-path router query"
+
+```sql
+-- Fetches the count of users born in the same year, but only
+-- for a single country
+SELECT EXTRACT(YEAR FROM date_of_birth) as birth_year, COUNT(*)
+FROM users_table
+WHERE country_code = 'USA' AND user_id = 15
```

Comment:
This query does not concur with the comment above (Fetches the count of users born in the same year, but only  for a single country) . There is only 1 user with user_id=15.